### ISSUE TITLE
TRAVEL_TOO_FAST validation check results in false positives

### ISSUE THREAD

**Issue by landonreed** [conveyal/gtfs-lib#303]
## Observed behavior

When validating a GTFS file, the `TRAVEL_TOO_FAST` check in `SpeedTripValidator.java` is returning false positives. It appears to be happening for trips with sets of stop_times that have no arrival/departure times. The issue can be seen with this 
[VTA GTFS feed](https://github.com/conveyal/gtfs-lib/files/6029727/VTA-20210218T193307Z-04a37fbe-f86c-4f0a-ad55-a3d51ab5315a-7d464169-97b4-497b-93b0-c68f28e4559f.1.zip).

https://github.com/conveyal/gtfs-lib/blob/3250327c3291c81d654e81c00ea1807791b0ba2f/src/main/java/com/conveyal/gtfs/validator/SpeedTripValidator.java#L86-L88

## Expected behavior

The `TRAVEL_TOO_FAST` error should only be flagged when the travel is actually too fast. For ranges of stop_times that do not have arrival/departure values, Euclidean distance (straight line distance) between the range of stops should be calculated and the average travel speed should be calculated and checked.

## Steps to reproduce the problem

1. Run `GTFS#main` with `--load /path/to/VTA-gtfs.zip --validate`
2. Open errors table of loaded feed to inspect TRAVEL_TOO_FAST items.

## Notes on configuration and files required to reproduce the issue

n/a

**Comment by br648** [comment]
@landonreed. The distance between a range of stops when no arrival/departure values are available is calculated incorrectly. For the test case, I have plotted the stops on Google maps and the total distance is 3.23km. The speed validation has this at 14.9km, hence the high speed. The cumulative distance (for each stop with no arrival/departure) is calculated from the current stop back to the last know previous stop with an arrival/departure time instead of the previous stop. e.g.

Stop1 (Previous stop with arrival/departure): 0
Stop 2 distance from stop 1: 100
Stop 3 distance from stop 1: 200
Stop 4 distance from stop 1: 300
Incorrect cumulative distance (as currently calculated): 100 + 200 + 300 = 600
Correct cumulative distance: 100 + 100 + 100 = 300

Should be a case of handing the previous stop with arrival/departure separately from previous stop for distance.

BTW, I tested the method `Util#fastDistance` against https://www.geodatasource.com/developers/java and the difference was less than a meter, so I'm happy that individual distances are being calculated correctly.
